Dockerfile

```diff
@@ -39,31 +36,158 @@ RUN conda install --name python2 --quiet --yes \
-RUN pip install https://github.com/IDR/nbresuse/archive/0.1.0-idr.zip && \
-    jupyter serverextension enable --py nbresuse && \
-    jupyter nbextension install --py --user nbresuse && \
-    jupyter nbextension enable --py --user nbresuse
+        idr-py==0.1.1
 
 RUN mkdir -p /home/jovyan/.local/share/jupyter/kernels/python2 && \
     sed 's/Python 2/OMERO Python 2/' \
         /opt/conda/envs/python2/share/jupyter/kernels/python2/kernel.json > \
         /home/jovyan/.local/share/jupyter/kernels/python2/kernel.json
 
-# Don't rename user- causes problems with earlier hardcoded paths
-#RUN usermod -l omero jovyan -m -d /home/omero
-#USER omero
+# Install git and pull the notebooks from the training repository
+RUN conda install --name python2 --quiet --yes -c anaconda git
+
+# Switch to root user for installing Cell Profiler dependencies
+USER root
+
+# Install CellProfiler dependencies
+RUN   apt-get -y update &&                                          \
+      apt-get -y install                                            \
+        build-essential    \
+        cython             \
+        git                \
+        libmysqlclient-dev \
+        libhdf5-dev        \
+        libxml2-dev        \
+        libxslt1-dev       \
+        openjdk-8-jdk      \
+        python-dev         \
+        python-pip         \
+        python-h5py        \
+        python-matplotlib  \
+        python-mysqldb     \
+        python-scipy       \
+        python-numpy       \
+        python-wxgtk3.0    \
+        python-zmq
+
+WORKDIR /usr/local/src
+
+# Install CellProfiler
+RUN /opt/conda/envs/python2/bin/git clone https://github.com/CellProfiler/CellProfiler.git
+
+WORKDIR /usr/local/src/CellProfiler
+
+ARG version=3.0.0
+
+RUN /opt/conda/envs/python2/bin/git checkout tags/v$version
+
+RUN /opt/conda/envs/python2/bin/pip install --editable .
+
+# R Installation (Staying as root user)
+RUN apt-get update -y && \
+    apt-get install -y nodejs wget git
+
+## Install R tools
+RUN /opt/conda/envs/python2/bin/conda config --add channels bioconda && \
+    /opt/conda/envs/python2/bin/conda install --quiet --yes \
+    'r-mclust' \
+    'r-ggdendro' \
+    'r-igraph' \
+    'r-pheatmap'
 
+# Dependencies necessary for install.R
+RUN echo "deb-src http://deb.debian.org/debian testing main" >> /etc/apt/sources.list
+RUN echo "deb http://ftp.debian.org/debian jessie-backports main" >> /etc/apt/sources.list
+RUN apt-get update && \
+    apt-get -y install libssl-dev libxml2-dev libcurl4-openssl-dev
+
+#########################################################################################
+## Necessary for running maven
+## copied from https://hub.docker.com/r/cardcorp/r-java/~/dockerfile/
+##
+## gnupg is needed to add new key
+RUN apt-get update && apt-get install -y gnupg2
+
+## Install Java 
+RUN apt-get update \
+    && apt-get install -t jessie-backports -y openjdk-8-jdk \
+    && echo "MAVEN IS NOT IN THE UPSTREAM LIST (JOSH)" \
```

Comment:
Which version was this written for? `apt-get install openjdk-8-jdk maven` seems to work

Reply:
@joshmoore : I did not change too many of these steps, when I tried to recover your branch. Happy to discuss and reduce the complexity if need be.